File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/util/validator/CEP.java
package br.edu.utfpr.serverpedidosjava.util.validator;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

@Documented
@Constraint(validatedBy = CEPValidator.class)
@Target({ ElementType.FIELD, ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)
public @interface CEP {

    String message() default "{cep.invalido}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/util/validator/CEP.kt
package br.edu.utfpr.serverpedidos.util.validator

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [CEPValidator::class])
@Target(AnnotationTarget.FIELD, AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class CEP(
    val message: String = "{cep.invalido}",
    val groups: Array<KClass<Any>> = [],
    val payload: Array<KClass<Payload>> = []
)
